### DISCUSSION TITLE
.net Maui: Show label or other information on a pin

### DISCUSSION THREAD

**Discussion by gfmoore** [Mapsui/Mapsui#1672] [Q&A]
Since this isn't a bug I've put this question here for now.

I have a a routine to add a pin to a map. This is all now working very well and I'm very pleased with the result. 

I would like to enhance the pin by showing some additional information. I note that there is a Label attribute (and an Address field?).

Is it possible to show this information as I add the pin, or otherwise enable it.

```
  public void AddPin(PostcodePosition p, Color c)
  {
    var myPin = new Pin(mapView)
    {
      Position = new Position(p.Latitude, p.Longitude),
      Type = PinType.Pin,
      Label = p.Postcode,
      Address = "",
      Scale = 0.7F,
      Color = c,
    };
    mapView.Pins.Add(myPin);
}
```

Thanks
Gordon

**Comment by charlenni** [comment]
It is shown in the samples.

Add an event when the Pin is clicked to the MapView

        mapView.PinClicked += OnPinClicked;

Then use this event to show the callout or hide it.

        private void OnPinClicked(object? sender, PinClickedEventArgs e)
        {
            if (e.Pin != null)
            {
                if (e.NumOfTaps == 2)
                {
                    // Hide Pin when double click
                    e.Pin.IsVisible = false;
                }
                if (e.NumOfTaps == 1)
                    if (e.Pin.Callout.IsVisible)
                        e.Pin.HideCallout();
                    else
                        e.Pin.ShowCallout();
            }

            e.Handled = true;
        }

For further information: see [here](http://mapsui.com/api/Mapsui.UI.Forms.Pin.html) for Pin and [here](http://mapsui.com/api/Mapsui.UI.Forms.Callout.html) for Callout.

**Reply by gfmoore** [reply]
Thanks for that. I had seen the code and was going to have a go, but Is it possible to have the label visible as I add the pin? What is the Address field for?

**Reply by gfmoore** [reply]
Just tested, your code works on tap and double tap. That's a work around. Can I programmatically call a tap, my C# isn't that great (as you've probably realised, but I'm getting there :) )

**Comment by gfmoore** [comment]
Figured it out :)

So for benefit of others:

```
    var myPin = new Pin(mapView)
    {
      Position = new Position(p.Latitude, p.Longitude),
      Type = PinType.Pin,
      Label = p.Postcode,
      Address = "1",
      Scale = 0.7F,
      Color = c,
    };
    myPin.ShowCallout();
    myPin.Callout.ArrowHeight = 10;
    myPin.Callout.TitleFontSize = 14;
    myPin.Callout.Color = c;
    myPin.Callout.TitleFontColor = c;
    mapView.Pins.Add(myPin);
```
Can this code be improved?

Note also that there are many other properties. However, banging my old drum again, the documentation in the api does not contain even a basic description of what many of these properties do. For instance what is the difference between title and subtitle? A user of this (and I'll say it "great") package shouldn't have to do trial and error to figure it out.  And honestly some screen grabs would be enormously useful. :)

And I'll say it again, this package has the potential to be absolutely fabulous, now I have some things working, it is proving to be very fast and capable. So thanks to the devs. :)

By the way I am using it with itinero for routing between postcodes and it is brilliant.

**Comment by gfmoore** [comment]
![image](https://user-images.githubusercontent.com/22768967/179616084-9f786cd1-11d0-4351-ba9d-64e45d2f626c.png)

Well I did say a screen grab would be useful...

**Comment by charlenni** [comment]
Regarding title and subtitle see [here](https://en.m.wikipedia.org/wiki/Subtitle_(titling)). The rest is normally self explaining. Or what didn‘t you find?

**Comment by gfmoore** [comment]
Haha,  no I know what a title is and a subtitle (and the body text). I wasn't sure what it meant in the context of a callout. I think I was being a little naive. I didn't realise that a callout had those features. As I say a picture really helps.  Must read the documentation. That made me laugh 😃

**Comment by charlenni** [comment]
Title is the bigger one which comes first. Subtitle isn‘t as big as title and below of the later. Try it, what should going wrong. The only, that could happen, is, that you see nothing :-)
